### PR TITLE
Add a linux version of the all-clusters-app demo

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

This PR adds a `linux` version of the `examples/all-clusters-app` (or `examples/wifi-echo` until #3653 lands).

The specific goal I'm pursuing here is to add a build with all clusters for writing cirque tests.

 #### Summary of Changes
* Add `examples/wifi-echo/linux`
* Move the `gen/` folder of `examples/wifi-echo/server/esp32` to `examples/wifi-echo/all-clusters-common` in order to get them to be used for both the linux and the esp32 versions.

### CODE REVIEW

**Comment by vivien-apple** [comment]
> Size increase report for "esp32-example-build" from [99aaed7](https://github.com/project-chip/connectedhomeip/commit/99aaed7c894eed93e9fbf8dbf64c2f3e9fa121fd)
> 
> File	Section	File	VM
> chip-wifi-echo.elf	.dram0.bss	0	512
> chip-wifi-echo.elf	.flash.rodata	368	368
> chip-wifi-echo.elf	.flash.text	216	216
> Full report output

It may seems weird to see some bloat report for the esp32. But it may be that more code is compiled. I need to investigate exactly why.

One thing I noticed is that I add to fix `clusters-callback-stubs.c` for the door-lock-server-logging callback that was normally here. And last time I tried with chip-tool, the `GetLogRecord` command was returning an error with the 'Unsupported Command' error code. So it looks like it was not compiled for some reasons.

**Review by andy31415 — APPROVED** [review]

File: examples/wifi-echo/linux/BUILD.gn [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+config("includes") {
+  include_dirs = [
+    ".",
+    "include",
+    "${chip_root}/src/app/util",
+  ]
+}
+
+executable("chip-tool-all-clusters-server") {
```
Comment (mspang) [inline examples/wifi-echo/linux/BUILD.gn:29]: Should this be "all-clusters-server" ?

I know we have chip-tool-server but I am also confused by that name.

**Review by mspang — APPROVED** [review]

**Review by jelderton — APPROVED** [review]

**Comment by vivien-apple** [comment]
In order to get the build to works fine I add to remove a few additional things. It was not happening while I was building on macOS as the linker may operate a bit differently there. But on Linux, a few undefined methods was preventing the build to link properly.

For references the method was `emberAfNetworkIndexFromEndpoint` which does not have any definition but only a declaration.
In this PR I removed the whole block of code that references it because the only code path that use this method is `emberAfScenesClusterClearSceneTableCallback`, and this method is getting removed by #3688 

The other code path that was preventing the build was `emberAfInvalidImageId` which is defined as an extern const but was never defined. I removed it, as well as most of the OTA related code paths. The reason for this cleanup is that the whole set of OTA related callbacks are going to be removed in #3464 but #3464 does not clean up the various OTA references in `af.h` and `af-types.h`

**Comment by vivien-apple** [comment]
Also I have added a task to `examples-linux-standalone.yaml` in order to build the all-clusters-app under Linux. Otherwise the build issue on Linux where only picked up while I was trying to add it to cirque. I don't think this is the goal of cirque tests to pick up build issues...

File: examples/all-clusters-app/all-clusters-common/BUILD.gn [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    "${chip_root}/src/app/util/ember-print.cpp",
+    "${chip_root}/src/app/util/message.cpp",
+    "${chip_root}/src/app/util/process-cluster-message.cpp",
+    "${chip_root}/src/app/util/process-global-message.cpp",
+    "${chip_root}/src/app/util/util.cpp",
+    "gen/call-command-handler.c",
+    "gen/callback-stub.c",
+    "gen/clusters-callback-stubs.c",
+  ]
+
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/all-clusters-common/BUILD.gn:62]: As a possible followup, worth checking whether adding `-ffunction-sections` to cflags and c++ flags fixes the linker issues....

File: src/app/util/attribute-storage.cpp [hunk 3]
```diff
@@ -773,9 +773,9 @@ static uint8_t findClusterEndpointIndex(EndpointId endpoint, EmberAfClusterId cl
         {
             break;
         }
-        epi = static_cast<uint8_t>(epi +
-                                           (emberAfFindClusterIncludingDisabledEndpointsWithMfgCode(
-                                                emAfEndpoints[i].endpoint, clusterId, mask, manufacturerCode) != NULL)
+        epi = static_cast<uint8_t>((epi +
```
Comment (bzbarsky-apple) [inline src/app/util/attribute-storage.cpp:776]: This does not look right; it looks like this got broken in #3681; thank you for catching it.  The right thing should be:
```
        epi = static_cast<uint8_t>(epi +
                                           ((emberAfFindClusterIncludingDisabledEndpointsWithMfgCode(
                                                emAfEndpoints[i].endpoint, clusterId, mask, manufacturerCode) != NULL)
                                       ? 1
                                       : 0));
```

That is, add either 1 or 0 to `epi` depending on the complicated condition.

File: examples/all-clusters-app/all-clusters-common/BUILD.gn (hunk 2 above)
Comment (vivien-apple) [reply examples/all-clusters-app/all-clusters-common/BUILD.gn:62]: I opened #3718 to not forgot to look at that.

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**